control-plane/src/modules/router.ts

```diff
@@ -348,4 +350,39 @@ export const router = s.router(contract, {
       body: result,
     };
   },
+  getDeploymentUploadDetails: async (request) => {
+    if (!UPLOAD_BUCKET) {
+      return {
+        status: 501,
+      };
+    }
+
+    const owner = await auth.jobOwnerHash(request.headers.authorization);
+    const { clusterId, serviceName } = request.params;
+
+    if (!owner || owner.clusterId !== clusterId || !owner.cloudEnabled) {
+      return {
+        status: 401,
+      };
+    }
+
+    const key = ulid();
+    return {
+      status: 200,
+      body: {
+        packageUploadUrl: await getPresignedURL(
+          UPLOAD_BUCKET,
+          clusterId,
+          serviceName,
+          `${key}-package`,
+        ),
+        definitionUploadUrl: await getPresignedURL(
+          UPLOAD_BUCKET,
+          clusterId,
+          serviceName,
+          `${key}-definition`,
```

Comment:
Upload the type definition separately (with the same key) so we can retrieve this without needing to fetch the entire service. I.e for building clients.